framework/src/common/errors.rs

```diff
@@ -71,3 +78,68 @@ pub enum NetBricksError {
     #[doc(hidden)]
     __Nonexhaustive,
 }
+
+#[macro_export]
+macro_rules! error_chain {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Error, true)
+    };
+    () => {};
+}
+
+#[macro_export]
+macro_rules! error_causes {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Error, false)
+    };
+    () => {};
+}
+
+#[macro_export]
+macro_rules! warn_chain {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Warn, true)
+    };
+    () => {};
+}
+
+#[macro_export]
+macro_rules! warn_causes {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Warn, false)
+    };
+    () => {};
+}
+
+/// Read a `failure` `Error` and print out the causes and a backtrace as
+/// `log::error`s
+#[doc(hidden)]
+pub fn print_chain(e: &Error, log_level: log::LevelFilter, with_backtrace: bool) {
```

Comment:
a couple improvements we should make here.

* not write multiple log statements per error. it's ok for console display. but once we start aggregating these across nodes, it will be kinda problematic.
* avoid all this processing if the logging level is turned off. as it is now regardless if logging is on or not, we will loop through the causes and resolve the backtrace.

I think need to restructure the helper a bit to generate the string separate from the logging. something like

```
warn!("{}", stringify(error))
```